### DISCUSSION TITLE
Error: Could not find the public.functionName() function in the schema cache

### DISCUSSION THREAD

**Discussion by mateosuarezdev** [supabase/supabase#6461] [Questions]
Hi! I'm trying to run a function but I'm getting that error:
`Could not find the public.get_by_distance() function in the schema cache`

With this hint:
`If a new function was created in the database with this name and parameters, try reloading the schema cache`

Hope you can help me with this one!

**Comment by steve-chavez** [comment]
@suarezlifestyle How are you calling the function? And what is the function definition?

**Reply by mateosuarezdev** [reply]
@steve-chavez 

In SQL editor I'm getting another error:
`structure of query does not match function result type`

 I'm not being able to make this function to work

```
CREATE OR REPLACE FUNCTION get_by_distance(userlocation point)
RETURNS TABLE
(
  v_id int, v_created_at timestamp, v_name text, v_category text, v_subcategory text, v_department text, v_geolocation point, distance float
)

LANGUAGE 'plpgsql'

AS
$BODY$
BEGIN
RETURN QUERY
SELECT listings."id", listings."created_at", listings."name", listings."category", listings."subcategory", listings."department", listings."geoloctaion", (userlocation <@> geoloctaion) as distance
FROM listings
WHERE distance < 20
ORDER BY distance;
END;
$BODY$
```

I'm trying to return a table using distance parameter from current user

**Comment by mateosuarezdev** [comment]
Now I changed some things and was able to run it without errors but it isn't returning anything (it's returning empty).
However, when I run the SELECT itself (only the query like this):
```
SELECT *, (point(-56.2133269, -34.8580424) <@> geoloctaion) as distance
FROM listings
WHERE (point(-56.2133269, -34.8580424) <@> geoloctaion) < 20
ORDER BY distance;
```

It returns the rows, however into the function it doesn't...

Updated code:

```
CREATE OR REPLACE FUNCTION get_by_distance(userlocation point)
RETURNS TABLE
(
  name text, category text, subcategory text, department text, geolocation point, distance float
)

AS
$BODY$
BEGIN
RETURN QUERY
SELECT listings."name", listings."category", listings."subcategory", listings."department", listings."geoloctaion", (userlocation <@> geoloctaion) as distance
FROM listings
WHERE distance < 20
ORDER BY distance;
END;
$BODY$ LANGUAGE plpgsql;
```

**Reply by steve-chavez** [reply]
See if the `listings` table has RLS enabled

**Reply by mateosuarezdev** [reply]
@steve-chavez 
RLS Disabled for that table

I'm calling it from inside SQL Editor now using this:
```
SELECT get_by_distance(point(-56.2134998,-34.8582221))
```

**Reply by steve-chavez** [reply]
Were you able to solve this? It seems you no longer have the `Could not find the public.functionName() function in the schema cache` error.

**Reply by etiennelunetta** [reply]
I'm encountering a similar bug. I have restarted the service and have checked for duplicate functions. I can run the function no-problem using the SQL editor, just not using supabasejs. I've also disabled RLS on the relevant tables, and changed the security definer to be that of the function author.

**Reply by steve-chavez** [reply]
@etiennelunetta Can you share the function signature(body not necessary) plus the JS snippet you're using?

**Reply by etiennelunetta** [reply]
@steve-chavez Thank you for the response. It looks like it was an intermittent issue. Just got back and while I'm fairly certain I didn't change anything, it seems to be working.
```
let { data, error } = await supabase
  .rpc('calc_distance', {
    start_date: today.local.start(),
    end_date: today.local.end()
  })
```
<img width="1361" alt="Screen Shot 2022-06-23 at 3 21 53 PM" src="https://user-images.githubusercontent.com/11713562/175425023-d07215d2-ca95-4c87-a80b-f43d86297716.png">

**Reply by sean-gribben** [reply]
FYI the solution is to clear the cache using 
```sql
SELECT pg_stat_clear_snapshot();
```

https://stackoverflow.com/a/75453583

**Comment by yang-zhang-work** [comment]
another possibility if none of the above works is to check the arguments and their values being passed into the function, for example, if the value is `undefined`, supabase would consider that argument not being passed and so the function becomes a different one.
